secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v3_1_10/domesticpayments/DomesticPaymentConsentsApiController.java

```diff
@@ -0,0 +1,133 @@
+
+    private final OBValidationService<OBWriteDomesticConsent4> domesticConsentValidator;
+
+    private final OBWriteDomesticConsentResponse5Factory consentResponseFactory;
+
+    private final FundsAvailabilityService fundsAvailabilityService;
+
+    private final OBWriteFundsConfirmationResponse1Factory fundsConfirmationResponseFactory;
+
+    public DomesticPaymentConsentsApiController(DomesticPaymentConsentStoreClient consentStoreApiClient,
+                                                OBValidationService<OBWriteDomesticConsent4> domesticConsentValidator,
+                                                OBWriteDomesticConsentResponse5Factory consentResponseFactory,
+                                                FundsAvailabilityService fundsAvailabilityService,
+                                                OBWriteFundsConfirmationResponse1Factory fundsConfirmationResponseFactory) {
+
+        this.consentStoreApiClient = consentStoreApiClient;
+        this.domesticConsentValidator = domesticConsentValidator;
+        this.consentResponseFactory = consentResponseFactory;
+        this.fundsAvailabilityService = fundsAvailabilityService;
+        this.fundsConfirmationResponseFactory = fundsConfirmationResponseFactory;
+    }
+
+    @Override
+    public ResponseEntity<OBWriteDomesticConsentResponse5> createDomesticPaymentConsents(OBWriteDomesticConsent4 obWriteDomesticConsent4,
+                String authorization, String xIdempotencyKey, String xJwsSignature, DateTime xFapiAuthDate, String xFapiCustomerIpAddress,
+                String xFapiInteractionId, String xCustomerUserAgent, String apiClientId, HttpServletRequest request, Principal principal) throws OBErrorResponseException {
+
+        logger.info("Processing createDomesticPaymentConsents request - consent: {}, idempotencyKey: {}, apiClient: {}, x-fapi-interaction-id: {}",
+                    obWriteDomesticConsent4, xIdempotencyKey, apiClientId, xFapiInteractionId);
+
+        domesticConsentValidator.validate(obWriteDomesticConsent4);
+
+        final CreateDomesticPaymentConsentRequest createRequest = new CreateDomesticPaymentConsentRequest();
+        createRequest.setConsentRequest(obWriteDomesticConsent4);
+        createRequest.setApiClientId(apiClientId);
+        createRequest.setIdempotencyKey(xIdempotencyKey);
+        createRequest.setCharges(calculateCharges(obWriteDomesticConsent4));
+
+        final DomesticPaymentConsent consent = consentStoreApiClient.createConsent(createRequest);
+        logger.info("Created consent - id: {}", consent.getId());
+
+        return new ResponseEntity<>(consentResponseFactory.buildConsentResponse(consent, getClass()), HttpStatus.CREATED);
+    }
+
+    private List<OBWriteDomesticConsentResponse5DataCharges> calculateCharges(OBWriteDomesticConsent4 obWriteDomesticConsent4) {
+        // TODO add some logic to apply charges to payments
+        return List.of();
+    }
```

Comment:
No charges implemented at the moment, as UK banks do not generally apply charges to domestic payments.

We can allow this behaviour to be customised by having a charges calculator interface, but I don't think this is required for the minimum viable product.